### PR TITLE
ENH: new methods added to `smash.factory.Net`

### PR DESCRIPTION
**Pull request by nghi-truyen**
New features are added to `smash.factory.Net` sub-package:
- Net.set_weight to set weights for the regionalization NN
- Net.set_bias to set biases for the regionalization NN
- Net.get_weight to get weights of the regionalization NN
- Net.get_bias to get biases of the regionalization NN
- Net.forward_pass to run a forward pass for the regionalization NN

Some tests are also added for the new methods

### CODE REVIEW

File: smash/factory/net/net.py [hunk 1]
```diff
@@ -603,8 +621,280 @@ def _fit_d2p(
...
+
+        >>> net.set_weight(random_state=0)
+        >>> net.get_weight()
+        [array([[ 0.05636498,  0.24847928,  0.11866093],
+                [ 0.05182664, -0.08815584,  0.16846401]])]
+        """
+
+        value, random_state = _standardize_set_weight_args(self, value, random_state)
+
+        if (random_state is not None) and (value is None):
```
[the remark this answers is not in the document]
Comment (nghi-truyen) [reply smash/factory/net/net.py:682]: Non..

File: smash/factory/net/_standardize.py [hunk 2]
```diff
@@ -108,6 +112,95 @@ def _standardize_set_trainable_args(net: Net, trainable: tuple | list) -> list:
...
+
+def _standardize_forward_pass_x(input_shape: tuple, x: np.ndarray) -> np.ndarray:
+    if isinstance(x, np.ndarray):
+        if x.shape != input_shape:
+            raise ValueError(
+                f"Invalid shape for input x. Could not broadcast input array "
+                f"from shape {x.shape} into shape {input_shape}"
+            )
+    else:
+        raise TypeError("Input x must be a Numpy array")
```
[the remark this answers is not in the document]
Comment (nghi-truyen) [reply smash/factory/net/_standardize.py:199]: Yes

File: smash/factory/net/_layers.py [hunk 3]
```diff
@@ -150,14 +139,15 @@ def __init__(
...
+        self.weight_shape = (neurons, input_shape[-1])
 
         self.bias = None
+        self.bias_shape = (1, neurons)
 
         self.kernel_initializer = kernel_initializer
         self.bias_initializer = bias_initializer
 
-    # TODO TYPE HINT: replace function by Callable
-    def _initialize(self, optimizer: function):  # noqa: F821
```
[the remark this answers is not in the document]
Comment (nghi-truyen) [reply smash/factory/net/_layers.py:160]: Ouais.. faut mettre callable plutôt que function pour que le ruff sort pas ce commentaire

**Comment by pag13** [comment]
These changes look ok for me

**Review by pag13 — COMMENTED** [review]
Changes ok for me, thanks.

**Review by pag13 — APPROVED** [review]

**Pull request merged**